text/0000-introduce-a-new-signing-mechanism.md

```diff
@@ -0,0 +1,568 @@
+* Gem certificates expire after 1 year by default. Once a certificate has expired, associated signatures can no longer be verified.
+
+([More details on gem signing’s current design can be found here](https://docs.google.com/document/d/1wUQR6Y0-6fopjh0ot-ByHXvgYI2Q2MCh3Z96WQcsU_Q/edit#)).
+
+This combination of factors means that both gem maintainers and gem users are disincentivized to either sign gems or verify gem signatures. Indeed, [Rubygem’s own security documentation](https://guides.rubygems.org/security/#general) notes that:
+
+> … this method of securing gems is not widely used. It requires a number of manual steps on the part of the developer, and there is no well-established chain of trust for gem signing keys. … The goal is to improve (or replace) the signing system so that it is easy for authors and transparent for users.
+
+In an ideal world, gem maintainers would find gem signing safe and trivial, so that they are inclined to sign their gems. Gem users would be able to rely on nearly all gems being signed, giving them increased confidence that they are using gems that have neither been falsified nor tampered with. In order to achieve the ideal world, we need an easier method of signing gems and verifying signatures that are valid forever.
+
+## Guide-level explanation
+
+This RFC introduces changes in technology and policy. To ease adoption, this RFC proposes three main rollout phases:
+
+1. **Opt-in build-time signing of gems.** This allows us to test, polish and harden signing and verifying without disrupting all existing workflows. During this phase maintainers will need to deliberately choose to sign their gems, and they can be nudged with a message to do so at build time. Similarly, users will need to deliberately verify gem signatures, and we can also nudge them to do so. The `gem cert` command will be deprecated.
+1. **Opt-out build-time signing of gems.** At this phase we will have gained experience and confidence with the new system. The RubyGems client will switch policy to make gem signing the default behavior, meaning that deliberate effort is needed to _not_ sign a gem. We expect this would create a rising tide of signed gems on rubygems.org as gem maintainers release new versions. Verification would happen automatically, but the lack of a signature would not be fatal. Gem authors would be able to set a policy that all their gem releases must be signed ([analogous to MFA configuration](https://guides.rubygems.org/mfa-requirement-opt-in/)). We would introduce a policy that the top most-downloaded gems must be signed ([analogous to the proposal for MFA](https://github.com/rubygems/rfcs/pull/36)). The `gem cert` command would be removed.
```

Comment:
I'm a bit confused about the policy that the gem authors can set. If the author is opting out of signing, is there any secure way to communicate this policy to users? It seems that an attacker would be motivated to make a fake policy opting a package out of signing. Maybe this could instead be left to users similar to in part 3?